Made the about us page dynamic. Loads details from a json file.

Kartikdot opened the pull request:
Team page; static data 
Fixes #47
Please review the changes. Gradle version has not been updated.

Kartikdot committed:
Made the about us page dynamic. Loads details from a json file.

Swati4star commented on app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java:
```diff
@@ -0,0 +1,98 @@
+package nsit.app.com.nsitapp;
```
Why have you changed the file name to AboutUs2? Let it be AboutUs only.

Swati4star commented on app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java:
```diff
@@ -0,0 +1,98 @@
+        aboutUsListView.setAdapter(mAdapter);
+        aboutUsListView.addFooterView(footerView);
+        TextView rep = (TextView) rootView.findViewById(R.id.GoToRepo);
+        TextView con = (TextView) rootView.findViewById(R.id.cont);
```
Make the variable name more informative. rep & con doesn't actually tell, what these are for.

Kartikdot replied:
These variable names were picked up from the previous code. I'll change them.

Swati4star commented on app/src/main/java/nsit/app/com/nsitapp/MainActivity.java:
```diff
@@ -61,6 +61,7 @@ protected void onCreate(Bundle savedInstanceState) {
         requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        appContext=getApplicationContext();
```
What is the need of this? I don't think you are using it anywhere.

Kartikdot replied:
This is being used to get the json file using the getResources() method in class utility.

Swati4star commented on app/src/main/java/nsit/app/com/nsitapp/aboutUs_loader.java:
```diff
@@ -0,0 +1,27 @@
+ * Created by Kartik Kwatra on 27-08-2017.
+ */
+
+public class aboutUs_loader extends AsyncTaskLoader<List<AboutUsMember>> {
```
Class names should be like this..
"AboutUsLoader"

Swati4star commented on app/src/main/res/layout/aboutus_activity.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
aboutUs is a fragment. Isn't it?

Swati4star commented:
The functionality is complete overall. But there are a few changes, that needs to be taken care of, before merging. Also, Can you explain the reason for adding so many Java files? BAsically, tell the functionality of each one.

Kartik2016 commented:
Thank you for the review. I have noted the changes to be made. 
On it! I'll also explain the functionality of the java files after making the required changes.

Kartikdot committed:
1. Renamed class files
2. Renamed the rep and con variables to goToRepo and contribute respectively
3. Renamed the layout files since they are fragments

Kartik2016 commented:
Made the changes as per the review.

4 new Java files have been added and 1 modified. Here i list the functionality of each.

1. AboutUs - This is the main fragment. It sets the layout, initializes the adapter for the listview. It also implements Loader callbacks and calls the initLoader() method.

2. AboutUsMember - This class defines the properties of a team member which will be displayed on the about us page. Objects of this class are used to populate the list view.

3. AboutUsLoader - This is a custom loader. It extends the AsyncTaskLoader. I made a custom loader instead of directly using AsyncTaskLoader since in case the user rotates the phone, a new instance of AsyncTask would be created and a new thread would be started. This would result in wastage of memory. Also, repeated changes in orientation might start too many threads resulting the app to crash.

4. AboutUsUtility - This class fetches the list of team members from the json file and generates an ArrayList of the type AboutUsMember. Functions of this class are called on a separate thread using the loader. 

5. MemberAdapter - This class populates the list view using the ArrayList of AboutUsMember.

Kartik2016 commented:
Fixes #47

Kartik2016 commented:
Please merge #53
